lib/torrent.js

```diff
@@ -334,14 +334,16 @@ class Torrent extends EventEmitter {
       dht: !this.private && this.client.dht,
       tracker: trackerOpts,
       port: this.client.torrentPort,
-      userAgent: USER_AGENT
+      userAgent: USER_AGENT,
+      lsd: this.client.lsd
     })
 
     this.discovery.on('error', (err) => {
       this._destroy(err)
     })
 
-    this.discovery.on('peer', (peer) => {
+    this.discovery.on('peer', (peer, source) => {
+      this._debug('peer %s discovered via %s', peer, source)
```

Comment:
This is a useful log. Thanks for adding!

Reply:
😊